src/go/rpk/pkg/config/weak.go

```diff
@@ -477,6 +479,16 @@ func (rpkc *RpkNodeConfig) UnmarshalYAML(n *yaml.Node) error {
 	rpkc.Tuners.BallastFilePath = string(internal.BallastFilePath)
 	rpkc.Tuners.BallastFileSize = string(internal.BallastFileSize)
 	rpkc.Tuners.WellKnownIo = string(internal.WellKnownIo)
+	if internal.CoresPerDedicatedInterruptCore != nil {
```

Comment:
@r-vasquez here I have a question from the rpk side. I couldn't find any place where we handle default values for the config properties which is why I implemented it like this.

I am not sure there is a better place or way?

Reply:
I see now that the way it's implemented this means that some of these values get printed into the config when doing something like enabling prod mode.

Idealy I'd just want to use these defaults internally and not have them leak out when not needed.

Reply:
> What do you mean by wanting the defaults internally? are they going to be modified by users at any point? 

They can be modified by the user yes. But what I want to avoid is that assuming the user hasn't changed a property is that when we write out the config (because for example rpk generates a default redpanda.yaml) I don't want it to end up in the config file.

So the states are basically:
 - unset
 - explicitly set to true
 - explicitly set to false

but if it's unset I don't want the default to end up in any config written by rpk.